### ISSUE TITLE
Add support for AdvancedMarker

### ISSUE THREAD

**Issue by halvtomat** [inocan-group/vue3-google-map#199]
The google maps Marker element is [deprecated as of February 2023](https://developers.google.com/maps/deprecations#googlemapsmarker_in_the_deprecated_as_of_february_2023), Advanced marker is the new better version.

Read more about Advanced marker [here](https://developers.google.com/maps/documentation/javascript/reference/advanced-markers)

**Comment by HusamIbrahim** [comment]
Thanks for the heads up! We'll try and work that into our plans.

**Comment by ScreamZ** [comment]
![Screenshot 2024-03-07 at 11 33 37](https://github.com/inocan-group/vue3-google-map/assets/6640835/9ac36819-8f98-43d0-a8fa-d4d1c5841644)

For more details

**Comment by Molanda** [comment]
I was able to use the `AdvancedMarkerElement` by adding `marker` to the `libraries` prop.

```
<GoogleMap :libraries="['marker'] ...
```

Once I did that, I was able to add the marker as follows (see the [Advanced Usage](https://www.npmjs.com/package/vue3-google-map#advanced-usage) section).

```
const marker = new this.refMap.api.marker.AdvancedMarkerElement({
  map: this.refMap.map,
  position: {
    lat: -25.363,
    lng: 131.044,
  },
});
```

@HusamElbashir  'marker' should be added to the list of libraries to avoid any TypeScript warnings...

```
libraries: {
  type: Array as PropType<("drawing" | "geometry" | "localContext" | "places" | "visualization" | "marker")[]>,
  default: () => ["places"],
},
```

Thank you for putting the component together!

**Comment by allaghi** [comment]
> I was able to use the `AdvancedMarkerElement` by adding `marker` to the `libraries` prop.
> 
> ```
> <GoogleMap :libraries="['marker']" ...
> ```
> 
> Once I did that, I was able to add the marker as follows (see the [Advanced Usage](https://www.npmjs.com/package/vue3-google-map#advanced-usage) section).
> 
> ```
> const marker = new this.refMap.api.marker.AdvancedMarkerElement({
>   map: this.refMap.map,
>   position: {
>     lat: -25.363,
>     lng: 131.044,
>   },
> });
> ```
> 
> @HusamElbashir `marker` should be added to the list of libraries to avoid any TypeScript warnings...
> 
> ```
> libraries: {
>   type: Array as PropType<("drawing" | "geometry" | "localContext" | "places" | "visualization" | "marker")[]>,
>   default: () => ["places"],
> },
> ```
> 
> Thank you for putting the component together!

Could you please show us the full example how you solve it ?

Thanks in advance.

**Comment by oawebdev** [comment]
> I was able to use the `AdvancedMarkerElement` by adding `marker` to the `libraries` prop.
> 
> ```
> <GoogleMap :libraries="['marker']" ...
> ```
> 
> Once I did that, I was able to add the marker as follows (see the [Advanced Usage](https://www.npmjs.com/package/vue3-google-map#advanced-usage) section).
> 
> ```
> const marker = new this.refMap.api.marker.AdvancedMarkerElement({
>   map: this.refMap.map,
>   position: {
>     lat: -25.363,
>     lng: 131.044,
>   },
> });
> ```
> 
> @HusamElbashir `marker` should be added to the list of libraries to avoid any TypeScript warnings...
> 
> ```
> libraries: {
>   type: Array as PropType<("drawing" | "geometry" | "localContext" | "places" | "visualization" | "marker")[]>,
>   default: () => ["places"],
> },
> ```
> 
> Thank you for putting the component together!

Hello! How to use it with MarkerCluster component?

**Comment by luisfrancajr** [comment]
Hello, is there any way to use MarkerCluster with advanced marker?

**Comment by Indre87** [comment]
Any updates on this with MarkerCluster?

**Comment by VictorBalbo** [comment]
I created a PR to add the new `AdvancedMarker` component: #263
To be honest the new component doesn't seem to bring relevant (the only thing I saw was the option to custumize the pin withoout actually replacing it by an image), and also removed alot of events and options that existed on the old component (like the property for animation, which now have to be done by hand).

**Comment by HusamElbashir** [comment]
Released in [v0.21.0](https://github.com/inocan-group/vue3-google-map/releases/tag/v0.21.0). Thanks to @VictorBalbo for the contribution.
